### PR TITLE
[update] faq.rst

### PR DESCRIPTION
**Pull request by isdanni**
# related issue
#99

### CODE REVIEW

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 1]
```diff
@@ -3,209 +3,176 @@
...
-":ref:`troubleshooting`."
+"**A:** Is your Python version >= 3.6, and 64-bit? See :ref:`troubleshooting`."
 msgstr ""
+"**A:** 你的 Python 版本是否 >= 3.6 并且是 64 位？请参阅 :ref:"
+"`troubleshooting` 。"
 
 #: ../../faq.rst:9
 msgid "**Q:** Do we have something like ``ti.pi``?"
-msgstr ""
+msgstr "**Q:** 我们有像 ``ti.pi`` 之类的工具吗？"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
msgstr "**Q:** 我们有提供像 ``ti.pi`` 之类的常量吗？"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 2]
```diff
@@ -3,209 +3,176 @@
...
 msgid ""
-"**A:** No, but you may use ``math.pi`` or ``numpy.pi`` instead. Taichi is"
-" able to bake in these constants during JIT, so your kernels incur no "
-"runtime cost."
+"**A:** No, but you may use ``math.pi`` or ``numpy.pi`` instead. Taichi is "
+"able to bake in these constants during JIT, so your kernels incur no runtime "
+"cost."
 msgstr ""
+"**A:** 没有，但你可以使用 ``math.pi`` 或 ``numpy.pi`` 代替。Taichi能够在 JIT "
+"期间处理这些常量，因此内核无需产生运行时成本。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"期间处理这些常量，因此内核不会承担运行时成本。"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 3]
```diff
@@ -3,209 +3,176 @@
...
 
 #: ../../faq.rst:13
 msgid ""
 "**Q:** How do I **force** an outermost loop to be serial, i.e. **not "
 "parallelized**?"
-msgstr ""
+msgstr "**Q:** 如何 **强制** 序列化最外层的循环，即 **不并行化**？"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
msgstr "**Q:** 如何 **强制** 串行执行最外层的循环，即 **不并行化**？"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 4]
```diff
@@ -3,209 +3,176 @@
...
+"        ..."
 
 #: ../../faq.rst:23
 msgid ""
-"**Q:** What's the most convinent way to load images / textures into "
-"Taichi tensors?"
-msgstr ""
+"**Q:** What's the most convinent way to load images / textures into Taichi "
+"tensors?"
+msgstr "**Q:** 将图像 / 纹理加载到Taichi张量中最方便的方式是什么？"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
msgstr "**Q:** 怎样才能最方便地将图像/纹理加载到 Taichi 张量中？"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 5]
```diff
@@ -3,209 +3,176 @@
...
 
 #: ../../faq.rst:29
 msgid ""
-"**A:** Yes, as long as that *package* provides an interface with "
-"``numpy``, see :ref:`other_python_packages`."
+"**A:** Yes, as long as that *package* provides an interface with ``numpy``, "
+"see :ref:`other_python_packages`."
 msgstr ""
+"**A:** 可以，只要那个 *包* 提供一个 ``numpy`` 接口，请参阅 ：ref： ref:"
+"`other_python_packages` 。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"**A:** 可以，只要那个包提供 ``numpy`` 接口，请参阅 :ref:"
"`other_python_packages` 。"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 6]
```diff
@@ -3,209 +3,176 @@
...
+"Or make use the extension library `Taichi THREE <https://github.com/taichi-"
+"dev/taichi_glsl>` to render images and update to GUI in real-time."
 msgstr ""
+"或者使用扩展库 `Taichi THREE <https://github.com/taichi-dev/taichi_glsl>`_ 来"
+"渲染图像并实时更新到 GUI。"
 
 #: ../../faq.rst:44
 msgid "**Q:** How do I declare a tensor with **dynamic length**?"
-msgstr ""
+msgstr "**Q:** 如何声明具有**动态长度** 的张量？"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
msgstr "**Q:** 如何声明具有 **动态长度** 的张量？"
```

**Review by archibate — COMMENTED** [review]
LGTMig!

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 7]
```diff
@@ -3,209 +3,176 @@
...
 
 #: ../../faq.rst:2
 msgid "Frequently asked questions"
-msgstr ""
+msgstr "常见问题"
 
 #: ../../faq.rst:5
 msgid "**Q:** Installing Taichi with ``pip``, complains ``package not found``."
-msgstr ""
+msgstr "**Q:** 用 ``pip`` 安装Taichi，报错 ``package not found`` 。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
msgstr "**Q:** 用 ``pip`` 安装 Taichi 时，报错 ``package not found`` 。"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 8]
```diff
@@ -3,209 +3,176 @@
...
+"`other_python_packages` 。"
 
 #: ../../faq.rst:31
 msgid ""
-"**Q:** Shall we add some handy functions like ``ti.smoothstep`` or "
-"``ti.vec3``?"
+"**Q:** Shall we add some handy functions like ``ti.smoothstep`` or ``ti."
+"vec3``?"
 msgstr ""
+"**Q:** 我们该添加一些像 ``ti.smoothstep`` 或 ``ti.vec3`` 的方便函数吗 ？"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"**Q:** 我们可以添加一些诸如 ``ti.smoothstep`` 或 ``ti.vec3`` 之类的便捷函数吗 ？"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 9]
```diff
@@ -3,209 +3,176 @@
...
 
 #: ../../faq.rst:33
 msgid ""
-"**A:** No, but we provide them in an extension library `Taichi GLSL "
-"<https://taichi-glsl.readthedocs.io>`, install it using:"
+"**A:** No, but we provide them in an extension library `Taichi GLSL <https://"
+"taichi-glsl.readthedocs.io>`, install it using:"
 msgstr ""
+"**A:** 不，但我们在扩展库 `Taichi GLSL <https://taichi-glsl.readthedocs."
+"io>`_ 中提供，请使用以下命令安装："
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"**A:** 不必要，因为我们在扩展库 `Taichi GLSL <https://taichi-glsl.readthedocs."
"io>`_ 中已经进行了提供，请使用以下命令安装："
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 10]
```diff
@@ -3,209 +3,176 @@
...
-msgstr ""
+msgstr "python -m pip install taichi_glsl"
 
 #: ../../faq.rst:39
 msgid ""
-"**Q:** How can I **render 3D results** without writing a ray tracer "
-"myself?"
-msgstr ""
+"**Q:** How can I **render 3D results** without writing a ray tracer myself?"
+msgstr "**Q:** 如果不自己编写光线跟踪器，如何 **渲染 3D 结果** ？"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
msgstr "**Q:** 如何不用自己编写光线跟踪，就能 **渲染出 3D 结果** ？"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 11]
```diff
@@ -3,209 +3,176 @@
...
 
 #: ../../faq.rst:42
 msgid ""
-"**A:** You may export it with :ref:`export_ply_files` so that you could "
-"view it in Houdini or Blender."
+"**A:** You may export it with :ref:`export_ply_files` so that you could view "
+"it in Houdini or Blender."
 msgstr ""
+"**A:** 你可以使用 :ref:`export_ply_files` 导出它，以便在Houdini或Blender中查"
+"看。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"**A:** 你可以使用 :ref:`export_ply_files` 导出模型，接着就可以在 Houdini 或 Blender 中查"
"看了。"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 12]
```diff
@@ -3,209 +3,176 @@
...
+msgstr "**Q:** 如何声明具有 **动态长度** 的张量？"
 
 #: ../../faq.rst:48
 msgid ""
-"**A:** What you want may be the ``dynamic`` SNode, a kind of sparse "
-"tensor, see :ref:`dynamic`."
+"**A:** What you want may be the ``dynamic`` SNode, a kind of sparse tensor, "
+"see :ref:`dynamic`."
 msgstr ""
+"**A:** 你想要的可能是 ``dynamic`` SNode，一种稀疏的张量，见 :ref:`dynamic`。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"**A:** 你想要的可能是 ``dynamic`` SNode，这是一种稀疏张量，详见 :ref:`dynamic`。"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 13]
```diff
@@ -3,209 +3,176 @@
...
 "Or simply allocate a dense tensor large enough, and another 0-D tensor "
-"``tensor_len[None]`` for length record. But in fact, the ``dynamic`` "
-"SNode could be slower than the latter solution, due to the cost of "
-"maintaining the sparsity information."
+"``tensor_len[None]`` for length record. But in fact, the ``dynamic`` SNode "
+"could be slower than the latter solution, due to the cost of maintaining the "
+"sparsity information."
 msgstr ""
+"或者简单地分配足够大的密集张量，再分配一个 0-D 张量 ``tensor_len[None]`` 用于"
+"长度记录。但事实上，因为维护稀疏信息的成本， ``dynamic`` SNode 可能比后者慢。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"或者简单地分配足够大的稠密张量，再分配一个 0D 张量 ``tensor_len[None]`` 用于"
"记录长度。不过事实上，由于维护稀疏信息的成本开销， ``dynamic`` SNode 可能比后者慢。"
``

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 14]
```diff
@@ -3,209 +3,176 @@
...
 
 #: ../../faq.rst:50
 msgid ""
 "**Q:** Can a user iterate over irregular topologies (e.g., graphs or "
 "tetrahedral meshes) instead of regular grids?"
 msgstr ""
+"**Q:** 用户能否在不规则拓扑（例如图形或四面体网格）而不是常规网格上迭代？"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"**Q:** 用户能否在不规则拓扑（例如曲线或四面体网格）而不是常规网格上迭代？"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 15]
```diff
@@ -3,209 +3,176 @@
...
 #: ../../faq.rst:52
 msgid ""
-"**A:** These structures have to be represented using 1D arrays in Taichi."
-" You can still iterate over them using ``for i in x`` or ``for i in "
+"**A:** These structures have to be represented using 1D arrays in Taichi. "
+"You can still iterate over them using ``for i in x`` or ``for i in "
 "range(n)``."
 msgstr ""
+"**A:** 这些结构必须使用Taichi中的 1D 数组表示。你仍可以使用 ``for i in x`` "
+"或 ``for i in range(n)`` 中迭代它们。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"**A:** 这些结构必须使用 Taichi 中的 1D 数组表示。接着你就可以通过 ``for i in x`` "
"或 ``for i in range(n)`` 迭代遍历它们了。"
```

File: locales/zh_CN/LC_MESSAGES/faq.po [hunk 16]
```diff
@@ -3,209 +3,176 @@
...
-#~ "using ``for i in x`` or ``for "
-#~ "i in range(n)``. However, at compile "
-#~ "time, there's little the Taichi compiler"
-#~ " can do for you to optimize it."
-#~ " You can still tweak the data "
-#~ "layout to get different runtime cache"
-#~ " behaviors and performance numbers."
-#~ msgstr ""
-
+"但是，在编译时，Taichi编译器几乎无法优化。但你仍然可以调整数据布局，以获得不"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/faq.po]:
```suggestion
"但是，在编译时，Taichi 编译器几乎无法对其优化。不过你仍可以通过调整数据布局，获得不"
```

**Review by ArkhamWJZ — COMMENTED** [review]
LGTM！+ A few nits ✨

**Review by ArkhamWJZ — APPROVED** [review]

**Pull request merged**